### DISCUSSION TITLE
request.input & request.only is undefined with GET requests

### DISCUSSION THREAD

**Discussion by yarapolana** [adonisjs/core#1713] [Help] [answered]
I just created two new projects and I was trying to get the values of my inputs but both are coming out as `undefined`. Did something change?

![Screenshot 2020-10-04 at 21 26 53](https://user-images.githubusercontent.com/19730118/95027507-92043400-0688-11eb-82fe-313be0253e81.png)

I noticed fold and ace in package.json came with older versions both 6.0.0 on contraire to another project that was `ace 6.9.4` and `fold 6.4.0`

`newly created`
<img width="325" alt="Screenshot 2020-10-04 at 21 29 34" src="https://user-images.githubusercontent.com/19730118/95027541-d8599300-0688-11eb-8b91-8dc2fe9dbb2e.png">

`few weeks old`
![Screenshot 2020-10-04 at 21 30 47](https://user-images.githubusercontent.com/19730118/95027551-f1624400-0688-11eb-91a1-4025d7b3e007.png)

**Comment by thetutlage** [comment]
Weird. Can you share how did you create the project? `npm init` or `yarn create`? 

To fix the main issue, you need to upgrade dependencies manually. The following command should do it

```ts
npm i @adonisjs/ace@alpha @adonisjs/fold@alpha @adonisjs/core@alpha
```

**Comment by yarapolana** [comment]
Strange indeed I used `yarn create` for the two new ones and `npx create` for the oldest one
updating the dependencies for the projects created with yarn did not work that was my first attempt, my second attempt was recreating the projects with `npx create` the dependencies came up to date but the bizarre issue with `request.input` ,`request.only`, `request.validate` coming `undefined` persists.

**Comment by yarapolana** [comment, marked as answer]
<img width="466" alt="Screenshot 2020-10-05 at 09 02 59" src="https://user-images.githubusercontent.com/19730118/95060332-adebf200-06e9-11eb-8e2a-7ccab32e77e7.png">

Though the request is there the values are still undefined.
Steps I took:
1- create a new project with yarn
2- create a route with request.input, request.only, request.validate
3- return the values or console log them - undefined
4- delete project
5- yarn cache clean

6- redo create project with yarn
7- same steps 2-3
8- delete node_modules
9- install dependencies again with yarn
10- update dependencies with yarn
11- steps 2-3
12- delete project

12- redo create project with npx
13- same steps 2-3
14- same result undefined

node v12.16.0
npm v6.14.8
yarn v1.22.5

**Reply by thetutlage** [reply]
Ahhh! I just saw that you are making a `GET` request. The body parser ignores GET requests by default. However, you can enable it inside the config file.`config/bodyparser.ts` add `GET` to `whitelistedMethods` array.

All these instances reminds me that docs needs to be better

**Reply by yarapolana** [reply]
Ahaa... I started questioning my dev skills for a bit 😅
Thanks for the whitelist tip!
